src/algorithms/string/knuth-morris-pratt/__test__/knuthMorrisPratt.test.js

```diff
@@ -2,6 +2,9 @@ import knuthMorrisPratt from '../knuthMorrisPratt';
 
 describe('knuthMorrisPratt', () => {
   it('should find word position in given text', () => {
+    expect(knuthMorrisPratt('', '')).toBe(0);
```

Comment:
I think this test is wrong. We should expect `-1` here instead of `0` because having zero as a result would mean that I may fetch empty word out of `string[0]`. But this is not true since the `string` itself is empty.